airflow/exceptions.py

```diff
@@ -239,6 +239,28 @@ def __init__(self, dag_run: DagRun, execution_date: datetime.datetime, run_id: s
             f"A DAG Run already exists for DAG {dag_run.dag_id} at {execution_date} with run id {run_id}"
         )
         self.dag_run = dag_run
+        self.execution_date = execution_date
+        self.run_id = run_id
+
+    def serialize(self):
+        cls = self.__class__
+        # Note the DagRun object will be detached here and fails serialization, we need to create a new
```

Comment:
```suggestion
        # Note the DagRun object will be detached here and fails serialization, we need to create a new one
```
Maybe worth explaining why it happens, and how creating a new object helps to deal with it (either in the code, or in the PR, or here)